pkg/elevator/internal/controllers/tenantobj_controller.go

```diff
@@ -95,80 +100,21 @@ func (r *TenantObjReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 func (r *TenantObjReconciler) reconcileTenantObj(
-	ctx context.Context, tenantObj *unstructured.Unstructured,
-	config catalogv1alpha1.VersionExposeConfig,
+	ctx context.Context, tenantObj, providerObj *unstructured.Unstructured,
+	statusFields []catalogv1alpha1.FieldPath,
 ) error {
-	desiredProviderObj := r.newProviderObject()
-	desiredProviderObj.SetName(tenantObj.GetName())
-	desiredProviderObj.SetNamespace(tenantObj.GetNamespace())
-
-	err := controllerutil.SetControllerReference(
-		tenantObj, desiredProviderObj, r.Scheme)
-	if err != nil {
-		return fmt.Errorf("set controller reference: %w", err)
-	}
-
-	// prepare config
-	statusFields, otherFields := elevatorutil.SplitStatusFields(config.Fields)
-
-	// Lookup current instance
-	currentProviderObj := r.newProviderObject()
-	err = r.Get(ctx, types.NamespacedName{
-		Name:      desiredProviderObj.GetName(),
-		Namespace: desiredProviderObj.GetNamespace(),
-	}, currentProviderObj)
-	if err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("getting %s: %w", r.ProviderGVK.Kind, err)
-	}
-
-	if errors.IsNotFound(err) {
-		// Create the Provider Obj
-		if err = elevatorutil.CopyFields(tenantObj, desiredProviderObj, otherFields); err != nil {
-			return fmt.Errorf("copy fields: %w", err)
-		}
-
-		if err = r.Create(ctx, desiredProviderObj); err != nil {
-			return fmt.Errorf("creating %s: %w", r.ProviderGVK.Kind, err)
-		}
-		return nil
+	if err := r.Patch(ctx, providerObj, client.Apply, elevatorutil.FieldOwner); err != nil {
```

Comment:
If I understand correctly, here you use `Patch` for the entire object instead of `Create`/`Update` logic just because for server-side apply. I am not a big fan of it.
1. This drops the readability of the code, and makes an assumption that everyone knows about Patch can also perform a `Create` in the background.
2. For implementing this, all the unit tests are dropped.
3. This leads inconsistency of our controller implementation, should we drop all create/update and use patch for every controller?

Also, have you tested locally? I have seen error like:
```
admission webhook "mtenantfakedb.kubecarrier.io" denied the request: webhook patch: Apply failed with 1 conflict: conflict with "before-first-apply" using eu-west-1.loodse/v1: .spec.image
```
when I tried to test it locally, and try to use `kubectl apply -f tenantobj.yaml` after I changed it.
Also, the following logs of catapult seem not good:
```
jiachengxu@jiachengs-mbp~/projects/github.com/kubermatic/kubecarrier (derivedCR-patch)$ kubectl logs internalfakedbs-eu-west-1-catapult-manager-74469c6f87-jwvll -n loodse
{"level":"info","ts":1584686815.148059,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1584686815.9521475,"logger":"controller-runtime.webhook","msg":"registering webhook","path":"/mutate-eu-west-1-loodse-v1-internalfakedb"}
{"level":"info","ts":1584686815.9525445,"logger":"catapult","msg":"starting manager"}
{"level":"info","ts":1584686815.9530926,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"info","ts":1584686815.953518,"logger":"controller-runtime.webhook.webhooks","msg":"starting webhook server"}
{"level":"info","ts":1584686816.0429776,"logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
{"level":"info","ts":1584686816.0432045,"logger":"controller-runtime.webhook","msg":"serving webhook server","host":"","port":9443}
{"level":"info","ts":1584686816.245382,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"internalfakedb","source":"kind source: eu-west-1.loodse/v1, Kind=InternalFakeDb"}
{"level":"info","ts":1584686816.342037,"logger":"controller-runtime.certwatcher","msg":"Starting certificate watcher"}
{"level":"info","ts":1584686816.343174,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"fakedb","source":"func source: 0x12d9e60"}
{"level":"info","ts":1584686816.643856,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"fakedb"}
{"level":"info","ts":1584686816.6439047,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"fakedb","worker count":1}
{"level":"info","ts":1584686816.7417963,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"internalfakedb","source":"func source: 0x12d9e60"}
{"level":"info","ts":1584686816.7418797,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"internalfakedb"}
{"level":"info","ts":1584686816.7419019,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"internalfakedb","worker count":1}
{"level":"info","ts":1584686894.643543,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate create","name":"db-1"}
{"level":"info","ts":1584686894.7452824,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate create","name":"db-1"}
{"level":"info","ts":1584686894.8363643,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate create","name":"db-1"}
{"level":"info","ts":1584686894.8428154,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate create","name":"db-1"}
{"level":"info","ts":1584686894.9398005,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate create","name":"db-1"}
{"level":"error","ts":1584686895.0567825,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"internalfakedb","request":"tenant/db-1","error":"creating FakeDb: fakedbs.stable.example.com \"db-1\" already exists","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/jiachengxu/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1584686896.2412446,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686896.4695902,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686896.538365,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686896.666447,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686896.7361927,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686896.7683823,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686896.8912797,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686897.116364,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686897.4885497,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686897.89403,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686898.2791522,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686898.7020445,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686899.0782368,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686899.4975796,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686899.9370627,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686900.2952845,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686900.9304197,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686901.1134372,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686901.4789133,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686901.894436,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686902.284694,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686902.703733,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686903.0836136,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686903.4945064,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686903.885304,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686904.2928817,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686904.6855628,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686905.0965443,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686905.4837184,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686905.8967526,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686906.2802148,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686906.7015095,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686907.0835924,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686907.4965034,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686907.8781562,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686908.291311,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686908.6813033,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686909.0931375,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686909.5443108,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686909.897376,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686910.2790132,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686910.69186,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686911.0761657,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686911.4948885,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686912.0364506,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686912.37922,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686912.6918378,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686913.0885458,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686913.4824984,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686913.9521275,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686914.2886877,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686914.6967797,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686915.0810692,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686915.4985518,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686915.883025,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686916.331149,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686916.6835973,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686917.1042197,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686917.484443,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686917.898463,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686918.2853487,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686918.663981,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686919.0452132,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686919.7184522,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686920.113065,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686920.4508476,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686920.9153135,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686921.2451081,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686921.66284,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686922.047829,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686922.4584274,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686922.8486912,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686923.301738,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686923.601514,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"error","ts":1584686923.6489463,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"internalfakedb","request":"tenant/db-1","error":"updating InternalFakeDb status: Operation cannot be fulfilled on internalfakedbs.eu-west-1.loodse \"db-1\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/jiachengxu/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1584686924.7312005,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686924.8015606,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686924.8442059,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686925.2615304,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686925.649431,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686926.0658164,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686926.449608,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686926.8597279,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686927.2514334,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686927.6844509,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686928.0488307,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686928.4724407,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686928.8467648,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686929.3156285,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686929.647465,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686930.2019875,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686930.5148785,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686930.8648026,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686931.4321375,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686931.7157204,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686932.0414648,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686932.4600794,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686932.8386288,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686933.2491593,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686933.6407886,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686934.0480547,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686934.4394257,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686934.8457265,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686935.240744,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686935.651765,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686936.03814,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686936.4430757,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686936.8404806,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686937.2470424,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686937.6472979,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686938.0482962,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686938.442458,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686938.8468037,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686939.2405972,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686939.6486082,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686940.0411248,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686940.450815,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686940.8391647,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686941.3060677,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686941.642633,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686942.0468233,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686942.4409292,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686942.846346,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"error","ts":1584686943.0433967,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"internalfakedb","request":"tenant/db-1","error":"updating InternalFakeDb status: Operation cannot be fulfilled on internalfakedbs.eu-west-1.loodse \"db-1\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/jiachengxu/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1584686944.0607095,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686944.0821092,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686944.4538043,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686944.6416757,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686945.0343885,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686945.4424074,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686945.8343716,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686946.256664,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686946.6322818,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686947.0437796,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686947.4353037,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686947.9010246,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686948.2377946,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686948.6280365,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686948.9963436,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686949.4128366,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686949.8011804,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686950.2061963,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686950.5994322,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686951.080054,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686951.399557,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686951.8109133,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686952.2043152,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686952.6388617,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686953.003327,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686953.410683,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686953.7977157,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686954.3744226,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686954.6332862,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686955.0038311,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686955.3972685,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686955.8326902,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686956.1967347,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686956.6208377,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686956.9986827,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686957.4014452,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686957.795496,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686958.201455,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686958.5959644,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686959.0021632,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686959.7695951,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686959.8675466,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686960.1935844,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686960.5990627,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686960.9964328,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686961.4028306,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686961.7948332,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686962.2021875,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686962.594037,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686963.0022447,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686963.4076083,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686963.803797,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686964.1978283,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686964.6125817,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686964.999171,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686965.7025824,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686965.834966,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686966.24865,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686966.5956697,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686967.0763164,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686967.3959773,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686967.805077,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686968.2111952,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686968.606953,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686969.0006225,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686969.4056082,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686969.796597,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686970.2124186,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686970.6016483,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686971.1687205,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686971.5661235,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686971.9141202,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686972.2008326,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686972.6122847,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686973.3668568,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686973.4663966,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686973.8006423,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686974.2628741,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686974.6012452,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686975.2157857,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686975.4548225,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686975.822133,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686976.201262,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686976.62032,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686977.0027175,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686977.4189796,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686977.8042107,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686978.3411741,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686978.571828,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686978.9833252,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686979.6318455,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686979.7909636,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686980.1690235,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686980.582635,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686980.9724147,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686981.4481335,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686982.0995793,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686982.307155,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686982.6496668,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686983.0318515,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686983.3628738,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686983.8363764,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686984.1615908,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686984.5791285,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686984.9649134,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686985.431571,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686985.6312602,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"error","ts":1584686985.834809,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"internalfakedb","request":"tenant/db-1","error":"updating InternalFakeDb status: Operation cannot be fulfilled on internalfakedbs.eu-west-1.loodse \"db-1\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/jiachengxu/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1584686986.8562124,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686986.8718982,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686987.043258,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686987.372692,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686987.7644951,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686988.1722798,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686988.563671,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686988.9993105,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686989.3611255,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686990.1331623,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686990.439286,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686990.761419,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686991.1689346,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686991.5609264,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686992.0311341,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686992.3627996,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686992.7730272,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686993.162958,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686993.5825307,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686993.9618523,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686994.3719268,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686994.7676105,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686995.182822,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686995.5891643,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686995.9731011,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686996.361828,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686996.7710364,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686997.1663103,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686997.577533,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686997.9848063,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686998.3751087,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686998.7652383,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686999.1732187,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686999.6345954,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584686999.9777644,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687000.3744407,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687000.7732322,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687001.1660168,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687001.5793629,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687001.9640572,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687002.4350848,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687002.7635171,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687003.2221568,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687003.430831,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"error","ts":1584687003.6349962,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"internalfakedb","request":"tenant/db-1","error":"updating InternalFakeDb status: Operation cannot be fulfilled on internalfakedbs.eu-west-1.loodse \"db-1\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/Users/jiachengxu/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/Users/jiachengxu/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/Users/jiachengxu/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1584687004.6640425,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687004.7322242,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687004.7650366,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687005.179641,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687005.5654783,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687006.0153506,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687006.3652246,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687006.7805974,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687007.1642256,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687007.580428,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687007.9667587,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687008.3823552,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687008.8021576,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687009.1429772,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687009.5984595,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687009.9962158,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687010.3333929,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687010.7435584,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687011.1319838,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687011.5580835,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687011.935663,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687012.3441336,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687012.744769,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687013.444571,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687013.5340385,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687013.9478962,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687014.336888,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687014.7488513,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687015.132032,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687015.596235,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687015.9371862,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687016.3469882,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687016.7305741,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687017.1589787,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687017.5373104,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687017.9448712,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687018.347464,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687018.7495797,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687019.138647,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687019.560761,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687019.93538,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687020.3484473,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687020.8113344,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687021.22583,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687021.5318882,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687021.9458735,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687022.3352866,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687022.8105106,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687023.1360786,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687023.549903,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687023.9371026,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687024.3409216,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687024.730556,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687025.148153,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
{"level":"info","ts":1584687025.5339608,"logger":"catapult.mutating webhooks.InternalFakeDb","msg":"validate update","name":"db-1"}
```
The TenantCRD object is keeping updating all the time, and I don't know why.

Reply:
Ok, I've improved our scenario test to include the updates and service cluster setting the status field.

The error is as follows; due to bug in Kubernetes server-side-apply status subresource handling, https://github.com/kubernetes/kubernetes/issues/88901, the `metadata.managedFields` is cleared upon a status update from the `CustromResourceDiscovery` controller which wrecks hell across the implementation. 

There could be a few possible workarounds, though I'm against using this feature until the status thing is properly fixed.